### PR TITLE
Prediction with Analytics View Models

### PR DESCRIPTION
**Pull request by hideaki**
# Description
Prediction with Analytics View Models.

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/build_coxph.R [hunk 1]
```diff
@@ -759,46 +774,78 @@ glance.coxph_exploratory <- function(x, data_type = "training", pretty.name = FA
...
+    ret <- tryCatch({
+      broom:::augment.coxph(x, data = NULL, newdata = cleaned_data, se = TRUE, ...)
+    }, error = function(e){
+      # TODO: This is copied from the code for glm, but check if this is useful for coxph.
+      # se=TRUE throws error that looks like "singular matrix 'a' in solve",
+      # in some subset of cases of perfect collinearity.
+      # Try recovering from it by running with se=FALSE.
+      broom:::augment.coxph(x, data = NULL, newdata = cleaned_data, se = FALSE, ...)
+    })
+    # TODO: Restore removed rows.
```
Comment (kei51e) [inline R/build_coxph.R:812]: Remove this comment.

File: R/randomForest_tidiers.R [hunk 2]
```diff
@@ -796,24 +796,31 @@ augment.ranger <- function(x, data = NULL, newdata = NULL, ...) {
 
 align_predictor_factor_levels <- function(newdata, model_df, predictor_cols) {
   cleaned_data <- newdata
-  # Align factor levels including Others and (Missing) to the model. TODO: factor level order can be different from the model training data. Is this ok?
+  # Align factor levels including Others and (Missing) to the model.
   for (i in 1:length(predictor_cols)) {
     predictor_col <- predictor_cols[i]
     training_predictor <- model_df[[predictor_col]]
-    if (is.factor(training_predictor) || is.character(training_predictor)) {
+    if ((is.factor(training_predictor) || is.character(training_predictor))) {
```
Comment (kei51e) [inline R/randomForest_tidiers.R:803]: Remove redundant parenthes.

File: R/randomForest_tidiers.R [hunk 3]
```diff
@@ -1255,16 +1273,22 @@ ranger.find_na_index <- function(variables, data) {
...
     predicted <- factor(levels_var[apply(pred, 1, function(x){
       if(is.na(x[2])){ # take care of the case where pred has only 1 column. possible when there are only one value in training data.
         1
       }
       else {
-        if(x[1]>threshold) 1 else 2
+        if (true_index == 1) {
+          if(x[1]>threshold) 1 else 2
+        }
+        else { # true_incex == 2
```
Comment (kei51e) [inline R/randomForest_tidiers.R:1289]:
```suggestion
        else { # true_index == 2
```

File: R/build_coxph.R (hunk 1 above)
Comment (hideaki) [reply R/build_coxph.R:812]: Done.

File: R/randomForest_tidiers.R (hunk 2 above)
Comment (hideaki) [reply R/randomForest_tidiers.R:803]: Done.

File: R/randomForest_tidiers.R (hunk 3 above)
Comment (hideaki) [reply R/randomForest_tidiers.R:1289]: Done in my commit to avoid unnecessary jenkins run.

**Pull request merged**